incubator/hnc/docs/user-guide/quickstart.md

```diff
@@ -0,0 +1,651 @@
+
+```bash
+kubectl -n team-b create role team-b-wizard --verb=update --resource=deployments
+kubectl -n team-b create rolebinding team-b-wizards --role team-b-wizard --serviceaccount=team-b:default
+```
+
+Now, if we assign the service to the new team:
+
+```bash
+kubectl hns set service-1 --parent team-b
+kubectl hns tree acme-org
+```
+
+Expected output:
+
+```
+acme-org
+├── team-a
+└── [s] team-b
+    └── service-1
+```
+
+And we can verify that the roles and rolebindings have been updated as well.
+
+```bash
+kubectl -n service-1 get roles
+kubectl -n service-1 get rolebindings
+```
+
+<a name="resources"/>
+
+## Propagating different resources
+
+_Demonstrates: simple HNC configuration._
+
+HNC doesn't only work for RBAC. Any Kubernetes resource can be configured to be
+propagated through the hierarchy, although by default, only RBAC objects are
+propagated.
+
+Continuing from the [previous quickstart](#basic), let's say that the workloads
+in `service-1` expect a secret called `my-creds` which is different for each
+team. Let's create those creds in `team-b`:
+
+```bash
+kubectl -n team-b create secret generic my-creds --from-literal=password=iamteamb
+```
+
+If you check existing secrets in `service-1` using the command below, you will
+find that the secret does not show up in `service-1` because we haven't
+configured HNC to propagate secrets in HNCConfiguration.
+
+```bash
+kubectl -n service-1 get secrets
+```
+
+In order to get this to work, you need to update the HNCConfiguration object,
+which is a single cluster-wide configuration for HNC as a whole. To do this,
+simply use the config subcommand:
+
+```bash
+# In HNC v0.6:
+kubectl hns config set-resource secrets --mode Propagate
+
+# In HNC v0.5:
+kubectl hns config set-type --apiVersion v1 --kind Secret propagate
+```
+
+  Note: In HNC v0.5, the apiVersion of _v1_ here applies to the Secret, and the word
+  _propagate_ refers to how we want HNC to treat this kind of object. In HNC
+  v0.6, the version is not required, but an optional `--group` parameter can
+  be used for resources that are not part of the core Kubernetes API.
+
+  Note: As of HNC v0.6, the supported modes are `Propagate`, `Remove` and
+  `Ignore`. More may be added in the future; see `kubectl hns config
+  set-resource` for the latest documentation.
+
+Now, we should be able to verify that `my-creds` was propagated to `service-1`:
+
+```bash
+kubectl -n service-1 get secrets
+```
+
+And finally, note that if we move the service back to team-a, the secret disappears because we haven't created it there:
+
+```bash
+kubectl hns set service-1 --parent team-a
+kubectl hns tree acme-org
+kubectl -n service-1 get secrets
+```
+
+If you like, you can also view the entire cluster-wide configuration for HNC by
+saying `kubectl hns config describe`. You can also look directly at the
+underlying cluster-wide configuration object:
```

Comment:
Maybe say "the entire cluster-wide configuration and **conditions**" now that we have the namespace conditions into the `conditions` field and they should show up in the `k hns config describe`.